sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1161,6 +1161,28 @@ class Dataset[T] private[sql](
       throw new AnalysisException("Invalid join type in joinWith: " + joined.joinType.sql)
     }
 
+    // If auto self join alias is enable
+    if (sqlContext.conf.dataFrameSelfJoinAutoResolveAmbiguity) {
```

Comment:
is this code repeated somewhere? and can we move it to a function to share code?

Reply:
yes, it's repeated from join function, I have created a new function and replaced duplicated code with it